simulator/src/models/power-plant.ts

```diff
@@ -0,0 +1,450 @@
+        } catch(err) {
+            console.trace(err);
+            return err;
+        }
+    }
+    
+
+    /**
+     * Gets the id of the power plant.
+     * @returns {uuid.v4} the power plants id.
+     */
+    get id(): uuid.v4 {
+        return this._id;
+    }
+
+
+    /**
+     * Gets the owner of the power plant.
+     * @returns {uuid.v4} the owner of the power plant.
+     */
+    get owner(): uuid.v4 {
+        return this._owner;
+    }
+    
+
+    /**
+     * Set the productionLevel variable.
+     * NOTE: Can't be lower then zero or higher the productionCapacity variable.
+     * @param {number} newProductionLevel the new value of the class variable productionLevel.
+     */
+    set productionLevel(newProductionLevel: number) {
+        if (newProductionLevel >= 0 && newProductionLevel <= this.productionCapacity) {
+            this._productionLevel = newProductionLevel;
+        }        
+    }
+
+
+    /**
+     * Set the productionRatio variable.
+     * @note Can't be lower the zero or higher then one;
+     * @param {number} newProductionRatio the new value of the class variable productionRatio.
+     */
+    set productionRatio(newProductionRatio: number) {
+        if (newProductionRatio >= 0 && newProductionRatio <= 1) {
+            this._productionRatio = newProductionRatio;
+        }
+    }
+
+
+    /**
+     * Simulates the amount of electricity produced.
+     * @param {Date} time the current time
+     * @returns {number} the simulated electricity production value.
+     */
+    private simProduction(time: Date): number {
+        return this._productionLevel + Math.sin(time.getTime()) * this.productionVariant;
+    }
+}
+
+
+/**
+ * Tells the state of object.
+ */
+enum Status {
+    Stopped,
+    Running,
+    Starting,
+    Stopping,
+}
+
+
+/***************************************************************************
+ * Production data useful for keeping history of wind speeds.
+ ***************************************************************************/
+
+
+/**
+ * PowerPlantData is a data structure for holding power plant data.
+ */
+export interface PowerPlantData {
+    readonly id: uuid.v4;
+    readonly time: Date;
+
+    readonly production: number;
+    readonly battery_value: number;
+    readonly unit: string;
+}
+
+
+/**
+ * Store a given power plant data in the power_plant_data table.
+ * If there already exists data for this time then update instead.
+ */
+async function storePowerPlantData(data: PowerPlantData) {
+    try {
+        await ElectricityGridDB.table('power_plant_data').insert_or_update(data, ['time']);
+    } catch (err) {
+        console.log("[Power Plant] Failed to store power plant data");
+    }
+}
+
+
+/**
+ * Get all data form a power plant.
+ */
+async function getPowerPlantData(id: uuid.v4): Promise<PowerPlantData[]> {
+    let queryText = `SELECT * FROM power_plant_data WHERE id = `;
+    let res = await ElectricityGridDB.query(queryText, [id]);
+    return res.rows;
```

Comment:
This can be done using ElectricityGridDB.table('power_plant_data').select([], [eq('id', id)])